pkg/scalers/kafka_scaler.go

```diff
@@ -211,24 +212,42 @@ func parseKafkaAuthParams(config *ScalerConfig, meta *kafkaMetadata) error {
 	}
 
 	if enableTLS {
-		certGiven := config.AuthParams["cert"] != ""
-		keyGiven := config.AuthParams["key"] != ""
-		if certGiven && !keyGiven {
-			return errors.New("key must be provided with cert")
-		}
-		if keyGiven && !certGiven {
-			return errors.New("cert must be provided with key")
-		}
-		meta.ca = config.AuthParams["ca"]
-		meta.cert = config.AuthParams["cert"]
-		meta.key = config.AuthParams["key"]
-		if value, found := config.AuthParams["keyPassword"]; found {
-			meta.keyPassword = value
-		} else {
-			meta.keyPassword = ""
+		if err := parseTLS(config, meta); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func parseTLS(config *ScalerConfig, meta *kafkaMetadata) error {
```

Comment:
moved this to a seperate function because of a linting error:
```
cyclomatic complexity 31 of func `parseKafkaAuthParams` is high (> 30) (gocyclo)
func parseKafkaAuthParams(config *ScalerConfig, meta *kafkaMetadata) error {
```